rules_128tech/pylint_plugins/mock_patch_autospec.py

```diff
@@ -38,32 +40,61 @@ def visit_call(self, call: astroid.node_classes.Call) -> None:
+        found_patch = False
+        expr = call.func
+        # Traverse the attribute looking for `patch`. This allows us to include
+        # mock.patch(), mock.patch.object(), mock.patch.multiple(), etc.
+        while isinstance(expr, astroid.node_classes.Attribute):
+            if expr.attrname == "patch":
+                found_patch = True
+
+            expr = expr.expr
+
+        if not found_patch or not isinstance(expr, astroid.node_classes.Name):
             return
 
-        if call.func.expr.name not in ("mock", "mocker"):
-            # Call to .patch() on some other object, I guess...
+        _parent_node, lookups = expr.lookup(expr.name)
+        for lookup in lookups:
+            if (
+                isinstance(lookup, astroid.node_classes.ImportFrom)
+                and lookup.modname == "unittest"
+            ):
+                self._check_patch_call(call)
+                return
+
+            if isinstance(lookup, astroid.node_classes.Import) and any(
+                orig_name == "unittest.mock" for orig_name, _alias in lookup.names
+            ):
+                self._check_patch_call(call)
+                return
+
+            # Pytest-mock fixtures are named "mocker" or "<scope>_mocker"
+            if isinstance(
+                lookup, astroid.node_classes.AssignName
+            ) and lookup.name.endswith("mocker"):
+                self._check_patch_call(call)
+                return
+
+    def _check_patch_call(self, call: astroid.node_classes.Call) -> None:
+        assert isinstance(call.func, astroid.node_classes.Attribute)
+
+        if call.func.attrname == "patch" and call.args and len(call.args) > 1:
+            # "new" is the second arg, and is incompatible with autospec
+            return
+        elif call.func.attrname == "object" and call.args and len(call.args) > 2:
+            # "new" is the third arg for patch.object()
             return
 
         if call.keywords:
             for keyword in call.keywords:
-                if keyword.arg == "autospec":
-                    if (
-                        isinstance(keyword.value, astroid.node_classes.Const)
-                        and keyword.value.value is True
-                    ):
-                        # They did the right thing!
-                        return
-
-                    # They passed a non-True value, probably an error
-                    self.add_message(
-                        "mock-patch-autospec", node=keyword.value,
-                    )
-
-        # We looked at all the kwargs and didn't find autospec=True, uh-oh
-        self.add_message("mock-patch-autospec", node=call)
+                if keyword.arg in ("new", "spec", "spec_set", "autospec"):
+                    # All of these args are incompatible with one another, as
+                    # long as one of them specified we are good
+                    return
+
+        # We looked at all the kwargs and didn't find 'autospec' or 'new', uh-oh
```

Comment:
nice, I didn't get a chance to reply to your comment about what to do when `autospec` is incompatible, but I agree that this is the right thing to do